interop-testing/src/test/java/io/grpc/testing/integration/ProxyTest.java

```diff
@@ -0,0 +1,233 @@
+package io.grpc.testing.integration;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketImpl;
+import java.net.UnknownHostException;
+import java.util.concurrent.TimeUnit;
+
+@RunWith(JUnit4.class)
+public class ProxyTest {
+
+  private int serverPort = 5001;
+  private int proxyPort = 5050;
+  private String loopBack = "127.0.0.1";
+
+  // These all work individually right now but running them sequentially causes errors
+  // because shutDown() is not clean. Holding off on fixing it right now because the threading
+  // model for the proxy is probably going to change (which will hopefully make shutting down
+  // easier).
+
+  @Test
+  @Ignore
```

Comment:
You can add a comment to ignores I thinkg:

`@Ignore("ignored until lorem ipsum")`